### PR TITLE
Switch from deprecated static-favicon to serve-favicon

### PR DESCRIPTION
**Pull request by totherik**
Closes #275. Will require update to [`generator`](https://github.com/krakenjs/generator-kraken/) to re-enable and configure correct favicon.

I'm not sure I love this fix, but for the short-term it should work. It entails:
- Update dependency to `serve-favicon`
- Disable favicon middleware by default.
- Configure favicon middleware to use icon provided by `kraken-js`, allowing applications to start. Unless enabled, this favicon will not be used out of the box.

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -40,11 +40,11 @@
...
 
         "favicon": {
-            "enabled": true,
+            "enabled": false,
             "priority": 30,
             "module": {
-                "name": "static-favicon",
-                "arguments": [ "path:./public/favicon.ico" ]
+                "name": "serve-favicon",
+                "arguments": [ "path:./node_modules/kraken-js/public/favicon.ico" ]
```
Comment (totherik) [inline config/config.json:47]: Looking for recommendations as to how best to avoid this.

Comment (aredridel) [inline config/config.json:47]: could be tidied with a shortstop config that does path resolution with module -- `resolve:kraken-js/public/favicon.ico` -- but still not super tidy.

Comment (jasisk) [inline config/config.json:47]: :+1: I can take care of that if so.

Comment (jasisk) [inline config/config.json:47]: `npm i shortstop-resolve`

**e**: #280

**Comment by jasisk** [comment]
Closing in favor of ~~mine~~ #280

**Pull request closed without merging**